subsys/fs/littlefs_fs.c

```diff
@@ -803,19 +830,30 @@ static int littlefs_mount(struct fs_mount_t *mountp)
 		LOG_INF("sizes: rd %u ; pr %u ; ca %u ; la %u",
 			lcp->read_size, lcp->prog_size, lcp->cache_size,
 			lcp->lookahead_size);
+#else
+		LOG_ERR("LittleFS disc access not enabled");
+		ret = -ENOSYS;
+		goto out;
+#endif
 	} else {
+#ifdef CONFIG_FS_LITTLEFS_FLASH_DEV
 		__ASSERT((((struct flash_area *)fs->backend)->fa_size %
 			  block_size) == 0,
 			 "partition size must be multiple of block size");
-		lcp->read = lfs_api_read;
-		lcp->prog = lfs_api_prog;
-		lcp->erase = lfs_api_erase;
+		lcp->read = lfs_api_read_flash;
+		lcp->prog = lfs_api_prog_flash;
+		lcp->erase = lfs_api_erase_flash;
```

Comment:
Why change names? They do not collide with anything.

Reply:
To have a naming pattern and future proofing the next memory access interface. Function without suffix sounds like an entry point for all other interfaces, but it isn't.

Reply:
But the commit does not add one, so no need to rename it for now.